packages/typescript-estree/src/convert.ts

```diff
@@ -3118,22 +3118,27 @@ export class Converter {
             node,
             this.ast,
           )!;
-          const withToken = findNextToken(openBraceToken, node, this.ast)!;
-          const withTokenRange = getRange(withToken, this.ast);
+          const withOrAssertToken = findNextToken(
+            openBraceToken,
+            node,
+            this.ast,
+          )!;
+          const withOrAssertTokenRange = getRange(withOrAssertToken, this.ast);
+          const withOrAssertName = withOrAssertToken.getText();
```

Comment:
> `token.getText()`

That's probably fine, but it would be a little more precise to check the `.kind`:

```suggestion
          const withOrAssertName =
            withOrAssertToken.kind === ts.SyntaxKind.AssertKeyword
              ? 'assert'
              : 'with';
```

IMO this is slightly preferable. But if another team member wants to come in and tell me `.getText()` is good enough and more succinct I won't complain.

Reply:
Addressed in 7af8e537c88e4166ca5b7bb6b8bc161fe1ed52db.

Since no-one else has (yet) expressed an opinion, I've gone with yours.